src/rgw/rgw_flight_frontend.cc

```diff
@@ -0,0 +1,224 @@
+
+  FlightFrontend::~FlightFrontend() {
+    delete flight_server;
+  }
+
+  int FlightFrontend::init() {
+    if (port <= 0) {
+      port = FlightServer::default_port;
+    }
+    const std::string url =
+      std::string("grpc+tcp://localhost:") + std::to_string(port);
+    flt::Location location;
+    arw::Status s = flt::Location::Parse(url, &location);
+    if (!s.ok()) {
+      ERROR << "couldn't parse url=" << url << ", status=" << s << dendl;
+      return -EINVAL;
+    }
+
+    flt::FlightServerOptions options(location);
+    options.verify_client = false;
+    s = flight_server->Init(options);
+    if (!s.ok()) {
+      ERROR << "couldn't init flight server; status=" << s << dendl;
+      return -EINVAL;
+    }
+
+    INFO << "FlightServer inited; will use port " << port << dendl;
+    return 0;
+  }
+
+  int FlightFrontend::run() {
+    try {
+      flight_thread = make_named_thread(server_thread_name,
+					&FlightServer::Serve,
+					flight_server);
+      set_flight_server(flight_server);
+
+      INFO << "FlightServer thread started, id=" <<
+	flight_thread.get_id() <<
+	", joinable=" << flight_thread.joinable() << dendl;
+      return 0;
+    } catch (std::system_error& e) {
+      ERROR << "FlightServer thread failed to start" << dendl;
+      return -ENOSPC;
+    }
+  }
+
+  void FlightFrontend::stop() {
+    set_flight_server(nullptr);
+    flight_server->Shutdown();
+    flight_server->Wait();
+    INFO << "FlightServer shut down" << dendl;
+  }
+
+  void FlightFrontend::join() {
+    flight_thread.join();
+    INFO << "FlightServer thread joined" << dendl;
+  }
+
+  void FlightFrontend::pause_for_new_config() {
+    // ignore since config changes won't alter flight_server
+  }
+
+  void FlightFrontend::unpause_with_new_config(rgw::sal::Store* store,
+			       rgw_auth_registry_ptr_t auth_registry) {
+    // ignore since config changes won't alter flight_server
+  }
+
+  /* ************************************************************ */
+
+  FlightGetObj_Filter::FlightGetObj_Filter(const req_state* request,
+					   RGWGetObj_Filter* next) :
+    RGWGetObj_Filter(next),
+    dp(request->cct->get(), dout_subsys, "rgw arrow_flight: "),
+    current_offset(0),
+    expected_size(request->obj_size),
+    uri(request->decoded_uri),
+    tenant_name(request->bucket->get_tenant()),
+    bucket_name(request->bucket->get_name()),
+    object_key(request->object->get_key()),
+    // note: what about object namespace and instance?
+    schema_status(arrow::StatusCode::Cancelled,
+		  "schema determination incomplete"),
+    user_id(request->user->get_id())
+  {
+#warning "TODO: fix use of tmpnam"
+    char name[L_tmpnam];
+    const char* namep = std::tmpnam(name);
+    if (!namep) {
+      //
+    }
+    temp_file_name = namep;
+
+    temp_file.open(temp_file_name);
+  }
+
+  FlightGetObj_Filter::~FlightGetObj_Filter() {
+    if (temp_file.is_open()) {
+      temp_file.close();
+    }
+    std::filesystem::remove(temp_file_name);
```

Comment:
`std::filesystem::remove()` throws on errors, which we don't want to do in a destructor. https://en.cppreference.com/w/cpp/filesystem/remove shows a non-throwing overload that takes a reference to `std::error_code`